test/lit_tests/raising/llvm_to_memref_access.mlir

```diff
@@ -0,0 +1,374 @@
+    return %1 : tensor<64xi64>
+  }
+}
+
+// CHECK-LABEL: func.func @multi_callers_kern
+// CHECK-SAME:  %[[ARG:.*]]: memref<64xi64, 1>
+// CHECK:       %[[ADDR:.*]] = "enzymexla.memref2pointer"(%[[ARG]]) : (memref<64xi64, 1>) -> !llvm.ptr<1>
+// CHECK:       llvm.getelementptr {{.*}} %[[ADDR]]
+
+// -----
+
+module {
+  llvm.func internal ptx_kernelcc @multi_callers_different_elty_kern(%arg0: !llvm.ptr<1>) {
+    %0 = llvm.mlir.constant(63 : i32) : i32
+    %1 = nvvm.read.ptx.sreg.tid.x : i32
+    %4 = llvm.zext %1 : i32 to i64
+    %5 = llvm.getelementptr inbounds %arg0[%4] : (!llvm.ptr<1>, i64) -> !llvm.ptr<1>, i64
+    %6 = llvm.load %5 {alignment = 1 : i64} : !llvm.ptr<1> -> i64
+    %7 = llvm.mul %6, %6 : i64
+    llvm.store %7, %5 {alignment = 1 : i64} : i64, !llvm.ptr<1>
+    llvm.return
+  }
+  func.func @main(%arg0: tensor<64xi64>) -> tensor<64xi64> {
+    %c0 = stablehlo.constant dense<0> : tensor<i64>
+    %c1 = stablehlo.constant dense<1> : tensor<i64>
+    %c40 = stablehlo.constant dense<40> : tensor<i64>
+    %0 = enzymexla.kernel_call @multi_callers_different_elty_kern blocks in (%c1, %c1, %c1) threads in (%c1, %c1, %c40) shmem=%c0 (%arg0) {output_operand_aliases = [#stablehlo.output_operand_alias<output_tuple_indices = [], operand_index = 0, operand_tuple_indices = []>]} : (tensor<64xi64>) -> tensor<64xi32>
+    %1 = enzymexla.kernel_call @multi_callers_different_elty_kern blocks in (%c1, %c1, %c1) threads in (%c1, %c1, %c40) shmem=%c0 (%0) {output_operand_aliases = [#stablehlo.output_operand_alias<output_tuple_indices = [], operand_index = 0, operand_tuple_indices = []>]} : (tensor<64xi32>) -> tensor<64xi64>
+    return %1 : tensor<64xi64>
+  }
+}
+
+// CHECK-LABEL: llvm.func internal ptx_kernelcc @multi_callers_different_elty_kern
+// CHECK-SAME:  %[[ARG:.*]]: !llvm.ptr<1>
+// CHECK-NOT:   "enzymexla.memref2pointer"
+// CHECK:       llvm.getelementptr {{.*}} %[[ARG]]
+
+// -----
+
+module {
+  llvm.func internal ptx_kernelcc @multi_callers_different_shape_kern(%arg0: !llvm.ptr<1>) {
+    %0 = llvm.mlir.constant(63 : i32) : i32
+    %1 = nvvm.read.ptx.sreg.tid.x : i32
+    %4 = llvm.zext %1 : i32 to i64
+    %5 = llvm.getelementptr inbounds %arg0[%4] : (!llvm.ptr<1>, i64) -> !llvm.ptr<1>, i64
+    %6 = llvm.load %5 {alignment = 1 : i64} : !llvm.ptr<1> -> i64
+    %7 = llvm.mul %6, %6 : i64
+    llvm.store %7, %5 {alignment = 1 : i64} : i64, !llvm.ptr<1>
+    llvm.return
+  }
+  func.func @main(%arg0: tensor<64xi64>) -> tensor<64xi64> {
+    %c0 = stablehlo.constant dense<0> : tensor<i64>
+    %c1 = stablehlo.constant dense<1> : tensor<i64>
+    %c40 = stablehlo.constant dense<40> : tensor<i64>
+    %0 = enzymexla.kernel_call @multi_callers_different_shape_kern blocks in (%c1, %c1, %c1) threads in (%c1, %c1, %c40) shmem=%c0 (%arg0) {output_operand_aliases = [#stablehlo.output_operand_alias<output_tuple_indices = [], operand_index = 0, operand_tuple_indices = []>]} : (tensor<64xi64>) -> tensor<8x8xi64>
```

Comment:
also can you add a test with a regular func call op with a tensor and inner stablehlo ops to make sure it doesn't crash on